src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,221 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node head;
+    private Node tail;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            tail = new Node(value);
+            head = tail;
+            wire(head, tail);
+        } else {
+            Node tmp = tail;
+            tail = new Node(value);
+            wire(tmp, tail);
+        }
+        size += 1;
+
+        return true;
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value,int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException(index);
+        }
+
+        Node target = getNodeByIndex(index);
+        Node newNode = null;
+
+        if (index == 0) {
+            head = new Node(value);
+            if (target == null) {
+                add(value);
+                return;
+            }
 
+            wire(head, target);
+        } else if (index == size) {
+            add(value);
+            return;
+        } else {
+            newNode = new Node(value);
+            wire(target.prev, newNode);
+            wire(newNode, target);
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T value,int index) {
+        validateIndex(index);
+        Node target = getNodeByIndex(index);
+        T result = target.value;
+        target.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        Node target = getNodeByIndex(index);
+        T result = target.value;
+        if (size == 1) {
+            tail = null;
+            head = null;
+        } else if (size == 2) {
+            if (target.value.equals(head.value)) {
+                head = tail;
+            } else {
+                tail = head;
+            }
+            wire(head, tail);
+        } else if (index == 0) {
+            head = target.next;
+            head.prev = head;
+        } else if (index == size - 1) {
+            tail = target.prev;
+            tail.next = null;
+        } else {
+            wire(target.prev, target.next);
+        }
+        size--;
+        return result;
     }
 
     @Override
     public boolean remove(T t) {
```

Comment:
in this method if else construction are too complicated, try to think how to simplify them